nixos/modules/services/desktops/pipewire.nix

```diff
@@ -25,17 +35,76 @@ in {
           Automatically run pipewire when connections are made to the pipewire socket.
         '';
       };
+
+      alsa = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Route audio to/from generic ALSA-using applications via the ALSA PIPEWIRE PCM plugin.
+          '';
+        };
+
+        support32Bit = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Whether to support sound for 32-bit ALSA applications on a 64-bit system.
+          '';
+        };
+      };
+
+      jack = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Enable transparent JACK audio emulation using LD_LIBRARY_PATH.
+          '';
+        };
+      };
+
+      pulse = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Enable transparent PulseAudio emulation using LD_LIBRARY_PATH.
+          '';
+        };
+      };
     };
   };
 
 
   ###### implementation
   config = mkIf cfg.enable {
-    environment.systemPackages = packages;
+    environment.systemPackages = [ pkgs.pipewire ]
+                                 ++ lib.optional cfg.jack.enable jack-libs
+                                 ++ lib.optional cfg.pulse.enable pulse-libs;
```

Comment:
Since filenames can conflict with pulseaudio and jack, I believe these should be at a lower priority.

Reply:
Filenames shouldn't conflict since they are put in a `pipewire` subdirectory.

Reply:
Ah sorry I read the derivation wrong.

Since you separated them, I thought you just put them in `$out/lib` instead of `$out/lib/pipewire`. Priorities and putting the libraries at `$out/lib` is one way to resolve the `$LD_LIBRARY_PATH` issue below though.

Reply:
The problem is that then the programs use the hard-coded rpath values before `systemPackages`. This means they find the original libraries instead of the replacements.